quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusFormatter.java

```diff
@@ -0,0 +1,173 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.redhat.quarkus.ls.commons.BadLocationException;
+import com.redhat.quarkus.ls.commons.TextDocument;
+import com.redhat.quarkus.model.Node;
+import com.redhat.quarkus.model.Node.NodeType;
+import com.redhat.quarkus.model.PropertiesModel;
+import com.redhat.quarkus.model.Property;
+import com.redhat.quarkus.settings.QuarkusFormattingSettings;
+import com.redhat.quarkus.utils.PositionUtils;
+
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4j.util.Ranges;
+
+/**
+ * Provides formatting support for an application.properties file
+ * 
+ * Formatting rules:
+ * Remove extra newlines in between properties
+ * Remove whitespaces before and after properties
+ * Add/remove spacing surrounding the equals sign, depending on <code>QuarkusFormattingSettings</code>
+ */
+class QuarkusFormatter {
+
+	private static final Logger LOGGER = Logger.getLogger(QuarkusFormatter.class.getName());
+
+	/**
+	 * Returns a <code>List<TextEdit></code> containing one <code>TextEdit</code> that formats the
+	 * the application.properties file represented by <code>document</code>.
+	 *
+	 * @param document           the properties model document
+	 * @param formattingSettings the client's formatting settings
+	 * @return a <code>List<TextEdit></code> that formats the
+	 * the application.properties file represented by <code>document</code>.
+	 */
+	public List<? extends TextEdit> format(PropertiesModel document, QuarkusFormattingSettings formattingSettings) {
+
+		Range fullRange = PositionUtils.createRange(document);
+		
+		if (fullRange == null) {
+			return null;
+		}
+		return format(document, fullRange, formattingSettings);
+	}
+
+	/**
+	 * Returns a <code>List<TextEdit></code> containing one <code>TextEdit</code> that formats the
+	 * the application.properties file represented by <code>document</code>, within the lines 
+	 * covered by the specified <code>range</code>.
+	 
+	 * @param document           the properties model document
+	 * @param range              the range specifying the lines to format
+	 * @param formattingSettings the client's <code>QuarkusFormattingSettings</code>
+	 * @return Returns a <code>List<TextEdit></code> that formats the
+	 * the application.properties file represented by <code>document</code>, within the lines
+	 * covered by the specified <code>range</code>.
+	 */
+	public List<? extends TextEdit> format(PropertiesModel document, Range range,
+			QuarkusFormattingSettings formattingSettings) {
+		
+		try {
+			enlargeRangeToGutters(range, document.getDocument());
+		} catch (BadLocationException e) {
+			LOGGER.log(Level.SEVERE, "Formatting failed due to BadLocation", e);
+			return null;
+		}
+
+		StringBuilder builder = new StringBuilder();
+
+		for (Node child: document.getChildren()) {
+			if (child.getNodeType() == NodeType.PROPERTY && isNodeInsideRange(child, range)) {
```

Comment:
You should take care of NodeType.COMMENTS because the format remove comments.

Reply:
thanks, my recent push takes care of it now